flink-kubernetes/src/test/java/org/apache/flink/kubernetes/utils/KubernetesUtilsTest.java

```diff
@@ -75,6 +81,91 @@ public void testCheckWithFixedPort() {
         testCheckAndUpdatePortConfigOption("6123", "16123", "6123");
     }
 
+    @Test
+    public void testLoadPodFromTemplateWithNonExistPathShouldReturnEmptyPod() {
+        final FlinkPod flinkPod =
+                KubernetesUtils.loadPodFromTemplateFile(
+                        flinkKubeClient,
+                        "/path/of/non-exist.yaml",
+                        KubernetesPodTemplateTestUtils.TESTING_MAIN_CONTAINER_NAME);
+        assertThat(flinkPod.getPod(), is(EMPTY_POD.getPod()));
+        assertThat(flinkPod.getMainContainer(), is(EMPTY_POD.getMainContainer()));
+    }
+
+    @Test
+    public void testLoadPodFromTemplateWithNoMainContainerShouldReturnEmptyPod() {
+        final FlinkPod flinkPod =
+                KubernetesUtils.loadPodFromTemplateFile(
+                        flinkKubeClient,
+                        KubernetesPodTemplateTestUtils.getPodTemplatePath(),
+                        "nonExistMainContainer");
+        assertThat(flinkPod.getMainContainer(), is(EMPTY_POD.getMainContainer()));
+        assertThat(flinkPod.getPod().getSpec().getContainers().size(), is(2));
+    }
+
+    @Test
+    public void testLoadPodFromTemplateAndCheckMetaData() {
+        final FlinkPod flinkPod =
+                KubernetesUtils.loadPodFromTemplateFile(
+                        flinkKubeClient,
+                        KubernetesPodTemplateTestUtils.getPodTemplatePath(),
+                        KubernetesPodTemplateTestUtils.TESTING_MAIN_CONTAINER_NAME);
+        assertThat(flinkPod.getPod().getMetadata().getName(), is("pod-template"));
```

Comment:
It is not really clear to my why the metadata name is `pod-template` if I only look at this test.

Reply:
I will add a comment here the `pod-template` is specified in `test-resources/testing-pod-template.yaml`.